light-client-verifier/src/verifier.rs

```diff
@@ -111,119 +111,163 @@ where
-    /// this will not (and will not be able to) check whether the untrusted
-    /// state's `next_validators_hash` field is valid.
-    fn verify(
+    /// Validates an `UntrustedBlockState`.
+    pub fn validate_untrusted(
         &self,
-        untrusted: UntrustedBlockState<'_>,
-        trusted: TrustedBlockState<'_>,
-        options: &Options,
-        now: Time,
-    ) -> Verdict {
-        // Ensure the latest trusted header hasn't expired
-        verdict!(self.predicates.is_within_trust_period(
-            trusted.header_time,
-            options.trusting_period,
-            now,
-        ));
-
-        // Ensure the header isn't from a future time
-        verdict!(self.predicates.is_header_from_past(
-            untrusted.signed_header.header.time,
-            options.clock_drift,
-            now,
-        ));
-
+        untrusted: &UntrustedBlockState<'_>,
+    ) -> Result<(), VerificationError> {
         // Ensure the header validator hashes match the given validators
-        verdict!(self.predicates.validator_sets_match(
+        self.predicates.validator_sets_match(
             untrusted.validators,
             untrusted.signed_header.header.validators_hash,
             &self.hasher,
-        ));
+        )?;
 
         // TODO(thane): Is this check necessary for IBC?
         if let Some(untrusted_next_validators) = untrusted.next_validators {
             // Ensure the header next validator hashes match the given next validators
-            verdict!(self.predicates.next_validators_match(
+            self.predicates.next_validators_match(
                 untrusted_next_validators,
                 untrusted.signed_header.header.next_validators_hash,
                 &self.hasher,
-            ));
+            )?;
         }
 
         // Ensure the header matches the commit
-        verdict!(self.predicates.header_matches_commit(
+        self.predicates.header_matches_commit(
             &untrusted.signed_header.header,
             untrusted.signed_header.commit.block_id.hash,
             &self.hasher,
-        ));
+        )?;
 
         // Additional implementation specific validation
-        verdict!(self.predicates.valid_commit(
+        self.predicates.valid_commit(
             untrusted.signed_header,
             untrusted.validators,
             &self.commit_validator,
-        ));
+        )?;
+
+        Ok(())
+    }
+
+    /// Verify that more than 2/3 of the validators correctly committed the block.
+    pub fn verify_commit(
+        &self,
+        untrusted: &UntrustedBlockState<'_>,
+    ) -> Result<(), VerificationError> {
+        self.predicates.has_sufficient_signers_overlap(
+            untrusted.signed_header,
+            untrusted.validators,
+            &self.voting_power_calculator,
+        )?;
+
+        Ok(())
+    }
+
+    /// Validate a `UntrustedBlockState`, based on the given `TrustedBlockState`, `Options` and
+    /// current time.
+    pub fn validate_trusting(
```

Comment:
```suggestion
    pub fn validate_unntrusted_against_trusted(
```

Reply:
I renamed `validate_trusting` to `validate_against_trusted` and `verify_commit_trusting` to `verify_commit_against_trusted` since I thought the `untrusted` part in the name was implicit, but please let me know if you prefer to keep it. 
https://github.com/informalsystems/tendermint-rs/pull/1226/commits/4555afee69b1dd4290fa644fd5ce747bd10f5452